src/hotspot/share/classfile/symbolTable.cpp

```diff
@@ -174,7 +174,8 @@ class SymbolTableConfig : public AllStatic {
       log_trace_symboltable_helper(&value, "Freeing permanent symbol");
       size_t alloc_size = SymbolTableHash::get_dynamic_node_size(value.byte_size());
       if (!SymbolTable::arena()->Afree(memory, alloc_size)) {
-        log_trace_symboltable_helper(&value, "Leaked permanent symbol");
+        // Can't print the now free'd symbol's value, but we just printed it above.
+        NOT_PRODUCT(log_trace(symboltable)(" - Leaked permanent symbol");)
```

Comment:
I think the comment may cause more confusion than its worth (the symbol hasn't been  "freed", or else there wouldn't be a leak). It's better to remove the comment.

Reply:
I modified the comment but still want it to ensure someone is not tempted to revert to the `log_trace_symboltable_helper` call - as it looks odd to use two different logging mechanisms so close together.